builder/vmware/iso/step_create_vmx.go

```diff
@@ -155,7 +156,34 @@ func (s *stepCreateVMX) Run(state multistep.StateBag) multistep.StepAction {
 	return multistep.ActionContinue
 }
 
-func (s *stepCreateVMX) Cleanup(multistep.StateBag) {
+func (s *stepCreateVMX) Cleanup(state multistep.StateBag) {
+	driver := state.Get("driver").(vmwcommon.Driver)
+	config := state.Get("config").(*Config)
+	ui := state.Get("ui").(packer.Ui)
+
+	ui.Say("Unregistering and deleting virtual machine...")
+	var err error = nil
+	for i := 0; i < 5; i++ {
+		running, err := driver.IsRunning(config.VMXTemplatePath)
+		if err == nil {
+			if running {
+				err = driver.Stop(config.VMXTemplatePath)
+			}
+		}
+		if err == nil {
```

Comment:
Sounds good to me, but would this mean we have to redo the code on other builders as well? I copied the "format" from the virtualbox builder.
Im about to update this request as i updated the code to be easier to read and fix some of the errors you mentioned. Let me know what you think of it, if you still feel we should abstract that functionality to its own function (because unit tests or otherwise)  im happy to do it.